gui/pages/swupd_config.go

```diff
@@ -0,0 +1,284 @@
+				page.mirrorWarning.SetText(utils.Locale.Get(swupd.IncorrectMirror)) // failure
+			} else {
+				page.model.SwupdMirror = mirror // success
+			}
+		}
+	}
+
+	page.setConfirmButton()
+}
+
+func (page *SwupdConfigPage) onAutoUpdateClick(button *gtk.CheckButton) {
+	var removeStyle, addStyle, warning string
+	if button.GetActive() {
+		removeStyle = "label-error"
+		addStyle = ""
+		warning = ""
+	} else {
+		removeStyle = ""
+		addStyle = "label-error"
+		warning = utils.Locale.Get(page.autoUpdateWarningMsg)
+	}
+
+	page.autoUpdateWarning.SetText(warning)
+	sc, err := page.autoUpdateWarning.GetStyleContext()
+	if err != nil {
+		log.Warning("Error getting style context: ", err) // Just log trivial error
+	} else {
+		sc.RemoveClass(removeStyle)
+		sc.AddClass(addStyle)
+	}
+}
+
+func (page *SwupdConfigPage) setConfirmButton() {
+	warning, err := page.mirrorWarning.GetText()
+	if err != nil {
+		log.ErrorError(err) // Just log trivial errors
+	}
+	if warning == "" {
+		page.done = true
+		page.controller.SetButtonState(ButtonConfirm, true)
+	} else {
+		page.done = false
+		page.controller.SetButtonState(ButtonConfirm, false)
+	}
+}
+
+// IsRequired will return false as we have default values
+func (page *SwupdConfigPage) IsRequired() bool {
+	return false
+}
+
+// IsDone checks if all the steps are completed
+func (page *SwupdConfigPage) IsDone() bool {
+	return page.done
+}
+
+// GetID returns the ID for this page
+func (page *SwupdConfigPage) GetID() int {
+	return PageIDConfigSwupd
+}
+
+// GetIcon returns the icon for this page
+func (page *SwupdConfigPage) GetIcon() string {
+	return "software-update-available-symbolic"
+}
+
+// GetRootWidget returns the root embeddable widget for this page
+func (page *SwupdConfigPage) GetRootWidget() gtk.IWidget {
+	return page.box
+}
+
+// GetSummary will return the summary for this page
+func (page *SwupdConfigPage) GetSummary() string {
+	return utils.Locale.Get("Software Updater Configuration")
+}
+
+// GetTitle will return the title for this page
+func (page *SwupdConfigPage) GetTitle() string {
+	return page.GetSummary()
+}
+
+// StoreChanges will store this pages changes into the model
+func (page *SwupdConfigPage) StoreChanges() {
+	page.validateMirror()
+	page.model.AutoUpdate = page.autoUpdateButton.GetActive()
+}
+
+// ResetChanges will reset this page to match the model
+func (page *SwupdConfigPage) ResetChanges() {
+	page.controller.SetButtonState(ButtonConfirm, true)
+	setTextInEntry(page.mirrorEntry, page.model.SwupdMirror)
+	page.autoUpdateButton.SetActive(page.model.AutoUpdate)
+}
+
+// GetConfiguredValue returns a string representation of the current config
+func (page *SwupdConfigPage) GetConfiguredValue() string {
+	var ret string
+	if page.model.AutoUpdate {
+		ret = utils.Locale.Get("Auto updates enabled")
+	} else {
+		ret = utils.Locale.Get("Auto updates disabled")
+	}
+
+	if page.model.SwupdMirror != "" {
+		ret += utils.Locale.Get(". Custom mirror set.")
```

Comment:
I'm not sure that it makes sense to have this leading person here for the previous statement.

Reply:
Done to have "." as a separate translated string.